armi/reactor/converters/uniformMesh.py

```diff
@@ -321,7 +344,15 @@ def applyStateToOriginal(self):
             f"Applying uniform neutronics mesh results from {self.convReactor} to ARMI mesh on {self._sourceReactor}"
         )
         completeStartTime = timer()
-        self._clearStateOnReactor(self._sourceReactor)
+
+        # Clear the state of the original source reactor to ensure that
+        # a clean mapping between the converted reactor for data that has been
+        # changed. In this case, we cache the original reactor's data so that
+        # after the mapping has been applied, we can recover data from any
+        # parameters that did not change.
+        self._cachedReactorCoreParamData = {}
```

Comment:
I'm a little worried that after this uniform mesh work is done, this dictionary will sit around forever with a reference to every Block in the Reactor.

This will amount to a memory leak, during long simulations.

After the mesh conversion is done, can we wipe out all the cache in this class?

Reply:
Yep! This is now updated. Please see the `reset()` method that is implemented to be called.